lib/puppet/provider/zone/solaris.rb

```diff
@@ -263,7 +263,8 @@ def start
 
       unless Puppet::FileSystem.exist?(sysidcfg)
         begin
-          File.open(sysidcfg, "w", 0600) do |f|
+          # For compatibility reasons use System encoding for this OS file
+          Puppet::FileSystem.open(sysidcfg, "w:#{Encoding.default_external.name}", 0600) do |f|
```

Comment:
Re: `sysidcfg` I was just going off the type reference, https://docs.puppet.com/puppet/latest/type.html#zone-attribute-sysidcfg. I was thinking since this will be coming in as a manifest value that it would be utf-8. Is that right?

I didn't know about `Encoding.find('filesystem')`. Cool.